src/chains/ethereum/transaction/tests/index.test.ts

```diff
@@ -1,6 +1,317 @@
+        assert.strictEqual(key, accessListStorageKey);
+      });
+      it("generates eip2930 access list transactions from raw buffer data", async () => {
+        const db: TypedDatabaseTransaction = txFromRpc.toEthRawTransaction(
+          txFromRpc.v.toBuffer(),
+          txFromRpc.r.toBuffer(),
+          txFromRpc.s.toBuffer()
+        );
+        const txFromDb = <EIP2930AccessListTransaction>(
+          TransactionFactory.fromDatabaseTx(db, common)
+        );
+        key = txFromDb.accessListJSON[0].storageKeys[0];
+        assert.strictEqual(txFromDb.type.toString(), "0x1");
+        assert.strictEqual(key, accessListStorageKey);
+      });
+
+      it("generates eip2930 access list transactions from raw string", async () => {
+        const str: string = "0x" + txFromRpc.serialized.toString("hex");
+        const txFromString = <EIP2930AccessListTransaction>(
+          TransactionFactory.fromString(str, common)
+        );
+        key = txFromString.accessListJSON[0].storageKeys[0];
+        assert.strictEqual(txFromString.type.toString(), "0x1");
+        assert.strictEqual(key, accessListStorageKey);
+      });
+    });
+  });
+
+  describe("LegacyTransaction Type", () => {
+    const rpc: TypedRpcTransaction = {
+      from: from,
+      to: to,
+      type: "0x0",
+      gasPrice: "0xffff"
+    };
+    const tx = <LegacyTransaction>TransactionFactory.fromRpc(rpc, common);
+    it("can be signed and hashed", () => {
+      // TODO: can we know the hash ahead of time if we supply the right data?
+      assert.strictEqual(typeof tx.hash, "undefined");
+      tx.signAndHash(fakePrivateKey);
+      assert.strictEqual(typeof tx.hash, "object");
+    });
+    describe("toVmTransaction", () => {
+      const vmTx = tx.toVmTransaction();
+      it("has a function to return the hash", () => {
+        assert.notDeepStrictEqual(vmTx.hash(), undefined);
+      });
+      it("has nonce property", () => {
+        assert.notDeepStrictEqual(vmTx.nonce, undefined);
+      });
+      it("has gasPrice property", () => {
+        assert.notDeepStrictEqual(vmTx.gasPrice, undefined);
+      });
+      it("has gasLimit property", () => {
+        assert.notDeepStrictEqual(vmTx.gasLimit, undefined);
+      });
+      it("has to property", () => {
+        assert.notDeepStrictEqual(vmTx.to, undefined);
+      });
+      it("has value property", () => {
+        assert.notDeepStrictEqual(vmTx.value, undefined);
+      });
+      it("has data property", () => {
+        assert.notDeepStrictEqual(vmTx.data, undefined);
+      });
+      it("has a function to get sender address", () => {
+        assert.notDeepStrictEqual(vmTx.getSenderAddress(), undefined);
+      });
+      it("has a function to get base fee", () => {
+        assert.notDeepStrictEqual(vmTx.getBaseFee(), undefined);
+      });
+      it("has a function to get base upfront cost", () => {
+        assert.notDeepStrictEqual(vmTx.getUpfrontCost(), undefined);
+      });
```

Comment:
I'd like exact values tested here, instead of `undefined`

Reply:
Updated to use exact values.